### ISSUE TITLE
Switch BatchNorm implementation to rely on the official tf.layers.batch_normalization

### ISSUE THREAD

**Issue by tvercaut** [NifTK/NiftyNet#285]
Batch Normalisation is officially supported in the main TensorFlow repo:
https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization

It would be beneficial to rely on this rather than our own implementation:
https://github.com/NifTK/NiftyNet/blob/dev/niftynet/layer/bn.py

Note that other users have done that already, see e.g.:
https://github.com/gift-surg/fetal_brain_seg/blob/master/NiftyNet/niftynet/layer/bn.py
vs.
https://github.com/gift-surg/fetal_brain_seg/blob/master/NiftyNet/niftynet/layer/bn_backup.py

**Comment by tvercaut** [comment]
This relates also to #280
